Feature: payout requests

JustSamuel opened the pull request:
# Description
This PR adds the creating and managing of `PayoutRequests` from the financial menu.

![image](https://github.com/user-attachments/assets/de5bd48d-112c-46dc-b38f-32ef3d384c9c)

The main table contains a tab for each of the 4 states. The actions allow a user to view a `PayoutRequest` or download the PDF.
The user can inspect the payout by pressing the eye icon:

![image](https://github.com/user-attachments/assets/4cb1cc07-5511-4e8b-802c-19cf704092c1)

This inspection allows the user to deny or approve the payout. When first inspected, the specific information such as the users' balance and payment info still has to be loaded and is displayed as a skeleton.

![image](https://github.com/user-attachments/assets/05aa1f3e-f908-4219-b238-b6246effec32)

After approving a payout, the table is reactively updated:

![image](https://github.com/user-attachments/assets/8209cccc-f53b-4923-96f8-74bc90a3d6e3)

Non `CREATED` payouts will have a slightly different inspection dialog. 

![image](https://github.com/user-attachments/assets/6cc5f287-7918-457d-89e2-451d285649bd)

An admin can also create a payout:

![image](https://github.com/user-attachments/assets/7f93e1c8-d188-4945-96c3-59854c4801d0)

The user selection dropdown uses lazy loading

The form has validation (notice how the submit button is disabled), and also shows the users current balance:

![image](https://github.com/user-attachments/assets/82eaf2c7-b9d9-4ca4-b944-d2e2271585f0)

Even a custom error when the balance of the user is lower than the requested payout:

![image](https://github.com/user-attachments/assets/07fb4251-d59a-49e3-af45-f131c86581a0)

After accepting, the form is reactively added to the table:

![image](https://github.com/user-attachments/assets/9964eabf-f0e7-4902-8841-5bafae54a179)


## TODO
- [x] translations
- [x] Refactor / clean up some code
- [x] Add loading indication when submitting form

## Types of changes
- New feature _(non-breaking change which adds functionality)_

JustSamuel committed:
feat: form utils and components

JustSamuel committed:
feat: stronger typing for form utils

JustSamuel committed:
feat: initial layout

JustSamuel committed:
feat: payout info modal

JustSamuel committed:
feat: payout info modal

JustSamuel committed:
feat: creating payouts

JustSamuel committed:
feat: small improvements

JustSamuel committed:
fix: added translations to PayoutInfo.vue

JustSamuel committed:
style: added comments and formatting

JustSamuel committed:
style: formatting and slight refactoring

JustSamuel committed:
fix: build errors (WIP)

JustSamuel committed:
feat: added notifications badge

JustSamuel committed:
chore: update client

JustSamuel committed:
chore: fixed build errors

JustSamuel committed:
fix: cleanup is array code

SuperVK commented:
![image](https://github.com/user-attachments/assets/ce4914b9-dded-4909-b56e-ee7b3c6a2fdd)
Might nice to add a hover tooltip for the right icon because it is not directly clear what it does. (I still don't know, because clicking it downloads a file without extension?)

![image](https://github.com/user-attachments/assets/6fe55ecb-8e87-4048-835a-a9b90429565b)
Capatilize pls :upside_down_face: 

Creating doesn't reactively add to the created list, in this out of the scope of this PR?

Viewing a created payout removes it from the list. And I cannot approve it either.


I am not sure if I like the tab view inside the cardpanel, it's out of line with the invoices and the mutations view. I get it because you need the button for creating, but maybe there is another option?

JustSamuel commented:
> Viewing a created payout removes it from the list. And I cannot approve it either.

> Creating doesn't reactively add to the created list, in this out of the scope of this PR?

Fixed, backend dev still had to be redeployed.

> Capatilize pls 🙃

fixed

JustSamuel commented:
@SuperVK 

![image](https://github.com/user-attachments/assets/05798bbd-d5ba-44e3-a679-be4b08b5a651)

opinion?

JustSamuel commented:
> Might nice to add a hover tooltip for the right icon because it is not directly clear what it does. (I still don't know, because clicking it downloads a file without extension?)

It downloads the PDF, however the file is extensionless (should be fixable in the proxy)

![image](https://github.com/user-attachments/assets/abf305a0-e581-4549-ac50-48a06d98d07a)

SuperVK commented:
> opinion?

Yeah I think it's better, not ideal tho, the spacing seems a bit funky. Maybe a red button? We will have the same problem very shortly when the "Create invoices" will appear in your invoice overview

JustSamuel committed:
fix(payout): i18n and some design fixes

JustSamuel commented:
> > opinion?
> 
> Yeah I think it's better, not ideal tho, the spacing seems a bit funky. Maybe a red button? We will have the same problem very shortly when the "Create invoices" will appear in your invoice overview

![image](https://github.com/user-attachments/assets/45b7d1bc-fcd2-40e3-b1be-8eb1894c0f96)

you could hack it inline?

JustSamuel committed:
fix: button styling

SuperVK commented:
The paginator is a bit broken, going back and forth doesn't really work, and the table is sorted ascending instead of descending (or the other way around, but not in the "normal" way atleast)

SuperVK commented:
Approving and denying payouts is also not reactively updated it seems

SuperVK commented:
> you could hack it inline?

I think that's the best option yeah

JustSamuel committed:
feat: updated reactivity
